src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/KeyworkerServiceController.kt

```diff
@@ -0,0 +1,291 @@
+    }
+
+    @ApiOperation(value = "Enable Manual Allocation and Migrate", notes = "Role Required: KW_MIGRATION. This will invoke migration from NOMIS DB")
+    @ApiResponses(value = [ApiResponse(code = 200, message = "OK", response = Prison::class), ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse::class), ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse::class)])
+    @PostMapping(path = ["/enable/{prisonId}/manual"])
+    fun addSupportedPrisonForManualAllocation(
+        @ApiParam("prisonId") @PathVariable("prisonId") prisonId: String,
+        @ApiParam("migrate") @Param("migrate") migrate: Boolean?,
+        @ApiParam(name = "capacity", value = "standard and extended default keyworker capacities for this prison, comma separated, e.g. &capacity=6,9") @Param("capacity") capacity: Array<Int>?,
+        @ApiParam(name = "frequency", value = "default KW Session Frequency in weeks (default 1)") @Param("capacity") frequency: Int?
+    ): Prison = updateAndMigrate(prisonId, migrate, false, capacity, frequency)
+
+    @ApiOperation(value = "Enable Auto Allocation for specified prison and Migrate", notes = "Role Required: KW_MIGRATION. This will also invoke migration from NOMIS DB")
+    @ApiResponses(value = [ApiResponse(code = 200, message = "OK", response = Prison::class), ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse::class), ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse::class)])
+    @PostMapping(path = ["/enable/{prisonId}/auto-allocate"])
+    fun addSupportedPrisonForAutoAllocation(
+        @ApiParam("prisonId") @PathVariable("prisonId") prisonId: String,
+        @ApiParam("migrate") @Param("migrate") migrate: Boolean?,
+        @ApiParam(name = "capacity", value = "standard and extended default keyworker capacities for this prison, comma separated, e.g. &capacity=6,9") @Param("capacity") capacity: Array<Int>?,
+        @ApiParam(name = "frequency", value = "default KW Session Frequency in weeks (default 1)") @Param("capacity") frequency: Int?
+    ): Prison {
+        return updateAndMigrate(prisonId, migrate, true, capacity, frequency)
+    }
+
+    private fun updateAndMigrate(prisonId: String?, migrate: Boolean? = false, autoAllocate: Boolean, capacity: Array<Int>?, kwSessionFrequencyInWeeks: Int? = 1): Prison {
+        if (capacity != null) {
+            Validate.isTrue(capacity.size == 2, "Two capacity values must be specified.")
+            prisonSupportedService.updateSupportedPrison(prisonId, autoAllocate, capacity[0], capacity[1], kwSessionFrequencyInWeeks)
+        } else {
+            prisonSupportedService.updateSupportedPrison(prisonId, autoAllocate)
+        }
+        migrate?.let { migrateBool ->
+            if(migrateBool) {
```

Comment:
this looks a lot messier than the original.  Why allow `migrate` to be null?

Reply:
because `migrate: Boolean?,` from the controller endpoint method is optional. Happy to go back to `if (migrate!!)` less noise in my view, and because it's defaulted in the method call it'll never throw an NPE

Reply:
done